qcodes/instrument_drivers/Keysight/private/Keysight_344xxA.py

```diff
@@ -0,0 +1,456 @@
+                           get_cmd='TRIGger:SLOPe?',
+                           vals=vals.Enum('POS', 'NEG'))
+
+        self.add_parameter('trigger_source',
+                           label='Trigger Source',
+                           set_cmd='TRIGger:SOURce {}',
+                           get_cmd='TRIGger:SOURce?',
+                           vals=vals.Enum('IMM', 'EXT', 'BUS', 'INT'))
+
+        # SAMPLING
+
+        self.add_parameter('sample_count',
+                           label='Sample Count',
+                           set_cmd=partial(self._set_databuffer_setpoints,
+                                           'SAMPle:COUNt {}'),
+                           get_cmd='SAMPle:COUNt?',
+                           vals=vals.MultiType(vals.Numbers(1, 1e6),
+                                               vals.Enum('MIN', 'MAX', 'DEF')),
+                           get_parser=int)
+
+        if DIG:
+            self.add_parameter('sample_count_pretrigger',
+                               label='Sample Pretrigger Count',
+                               set_cmd='SAMPle:COUNt:PRETrigger {}',
+                               get_cmd='SAMPle:COUNt:PRETrigger?',
+                               vals=vals.MultiType(vals.Numbers(0, 2e6-1),
+                                                   vals.Enum('MIN', 'MAX', 'DEF')),
+                               get_parser=int,
+                               docstring=('Allows collection of the data '
+                                          'being digitized the trigger. Reserves '
+                                          'memory for pretrigger samples up to the'
+                                          ' specified num. of pretrigger samples.')
+                               )
+
+        self.add_parameter('sample_source',
+                           label='Sample Timing Source',
+                           set_cmd='SAMPle:SOURce {}',
+                           get_cmd='SAMPle:SOURce?',
+                           vals=vals.Enum('IMM', 'TIM'),
+                           docstring=('Determines sampling time, immediate'
+                                      ' or using sample_timer'))
+
+        self.add_parameter('sample_timer',
+                           label='Sample Timer',
+                           set_cmd='SAMPle:TIMer {}',
+                           get_cmd='SAMPle:TIMer?',
+                           unit='s',
+                           vals=vals.MultiType(vals.Numbers(0, 3600),
+                                               vals.Enum('MIN', 'MAX', 'DEF')),
+                           get_parser=float)
+
+        self.add_parameter('sample_timer_minimum',
+                           label='Minimal sample time',
+                           get_cmd='SAMPle:TIMer? MIN',
+                           get_parser=float,
+                           unit='s')
+
+        # SYSTEM
+        self.add_parameter('error',
+                           label='Error message',
```

Comment:
according to `errorparser` implementation, this "Parameter" returns not only the message, but also the error code. In order to avoid confusion with the usage of this Parameter, and to avoid abuse of Parameter class as such, could you please make this a method of this class (basically making `errorparser` a method)?

Reply:
No problem